Ports/.port_include.sh

```diff
@@ -412,6 +429,13 @@ func_defined pre_configure || pre_configure() {
     :
 }
 func_defined configure || configure() {
+    if [[ -n "$(run find -maxdepth 1 -iname cmakelists.txt)" ]]; then
```

Comment:
I dislike all these repeated checks for a `cmakelists.txt` file; how about we do this properly and take a look at the `useconfigure` variable instead? We can probably do something like replace that with a configure system enum to switch between doing nothing, running the configure script and running cmake.

Reply:
I have hardly seen people using file name other than CMakeLists.txt in the project. So we can simplify it to

```bash
[[ -f "${workdir}/CMakeLists.txt" ]]
```

Reply:
I mean. `llvm` has the correct CMakeLists.txt in the `llvm/` sub-folder :shrug: but we can update this when we get to that port.

Reply:
I think my comment here was misunderstood.

Afaik, we use `useconfigure='true'` to use `./configure`-like scripts to set up the build system. The variable `configscript` can be set to change the filename for `configure`.

This PR however, sets up a CMake-style build which is entirely different. Instead of using autodetection of the configuration system, why not ask the port author to specify which configuration system to use exactly?

E.g. `configuration_type=configure`, `configuration_type=cmake`, etc.

Following that, you can always modify something like `configscript` to point to the right file and deal with `llvm` for eaxmple.

Reply:
I have understood your comment initially. When we initially run the cmake, the process is also called "configure the build" so I thought, if somehow we can use `useconfigure=true` which tells the script configure the build (run `do_configure`) before `do_build`.

Also one more question. How can I change global variable `workdir` in function and use it in different function.

This is currently happening in my case
```bash
workdir=src

pre_configure() {
  ...
  workdir+="/main_project"
}

configure() {
  echo "$workdir"   # src
}
``` 

But if I run this script out of `.port_include.sh`, it works.

Reply:
Hey @gmta while building some packages. I understood what you said and why it is required.
Basically useconfigure is used to enable configuring the port because building and installing it.

However I propose another variable name to do same thing `configure_tool`

```bash
configscript=configure
configure_tool=autoconf
configopts=()
useconfigure=false
```
This is how variables will look

Reply:
This is getting convoluted beyond maintainability.

Once I'm done with my current Ports patchset, I'll whip up a new one that just removes the whole thing in favor of a default-blank `configure()` method (even if just to see how it would look like).

I don't imagine it to be much worse (or any worse at all) than our current solution, or any other solutions that have been proposed.